modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/utils.ts

```diff
@@ -369,6 +369,16 @@ export function getUnsupportedObjectRelationshipErrorMessage(
 	}
 }
 
+function isTrustedURL(url: string): boolean {
+	const actualURL = new URL(window.location.href);
+	const targetURL = new URL(url);
+
+	return (
+		actualURL.origin === targetURL.origin &&
+		actualURL.hostname === targetURL.hostname
+	);
+}
+
 export function updatePreviousURLParam(paramType: string, paramValue: string) {
 	const previousPath = document.referrer;
```

Comment:
I believe that the big issue with this security vulnerability is the possibility of the user modifying the reference on the previous page. With your solution, you are no longer allowing the user to be redirected to another domain, which is certainly a significant improvement.

However, there is still the possibility that the user can add their expected parameter (`objectFolderName`) and be able to add unwanted information to the URL, since you only encode the parameters received from the function.

https://github.com/liferay-objects/liferay-portal/assets/61854510/db1ceb14-5be4-498f-b4a6-4667c79dc2c6